Makefile

```diff
@@ -122,7 +122,13 @@ cluster-clean:
 cluster-deploy: cluster-clean
 	./hack/cluster-deploy.sh
 
-cluster-sync: cluster-build cluster-deploy
+cluster-sync:
+ifeq ($(PARALLEL_CLEANUP), 1)
+	./hack/parallel.sh
+else
+	make cluster-build
+	make cluster-deploy
+endif
```

Comment:
@oshoval Since the parallelization is between build and clean, I would suggest to leave the cluster-sync target as-is and instead modify the cluster-build target, for example:

```
cluster-build:
ifeq ($(PARALLEL_CLEANUP), 1)
                ./hack/parallel.sh
else
                ./hack/cluster-build.sh
endif
```

In addition, I would remove the call to cluster-deploy.sh from the parallel.sh script, and let the cluster-deploy be called from the Makefile

Reply:
interesting, but i wanted to do a reuse of the current make targets, for example if their depend list would be change and so on. wdyt?

Reply:
@oshoval My only concern is calling make from inside a Makefile. Regarding cluster-deploy that depends on cluster-clean, we can make this dependency conditional using inline condition, for example:
```
cluster-deploy: $(if $(PARALLEL_CLEANUP),, cluster-clean)
```
For this sake we can sacrifice reuse a little I think.

Reply:
there isnt a concern at about running make from a makefile, please try and see,
its just as any other command, it runs as a separate instance, just as any other command,
even "regular" commands, run in a makefile, in a subshell of their own, this is why passing variables for example isnt straight forward between lines.

i think we should keep stuff simple and straight forward, readable and maintainable.